### PR TITLE
Add Youtube Music (Issue 805)

### PR DESCRIPTION
**Pull request by gapurov**
Added Youtube Music Media Strategy and adjusted Youtube Media Strategy to make them work together.
https://github.com/beardedspice/beardedspice/issues/805

### CODE REVIEW

**Comment by andreaswilli** [comment]
+1

File: BeardedSpice/MediaStrategies/Youtube.js [hunk 1]
```diff
@@ -4,16 +4,17 @@
...
 
 BSStrategy = {
-  version: 3,
+  version: 4,
   displayName: "Youtube",
   accepts: {
     method: "predicateOnTab",
-    format: "%K LIKE[c] '*youtube.com/watch*'",
-    args: ["URL"]
+    format: "%K LIKE[c] '*youtube.com/watch*' && !%@ LIKE[c] '*music.youtube.com*'",
```
Comment (davidstosik) [inline BeardedSpice/MediaStrategies/Youtube.js]: (I know nothing of Objective-C.) What is the difference between `%K` and `%@` in `NSPredicate`?

Figured I might as well Google it:
> - `%@` is a var arg substitution for an object value—often a string, number, or date.
> - `%K` is a var arg substitution for a key path.
>
> ([source](https://nshipster.com/nspredicate/))

(Doesn't makes things much clearer for me, to be honest. 😅 )

Seeing how `%K` is used in all strategies, would it make sense to keep it the same?

**Comment by andreaswilli** [comment]
@davidstosik Please have a look at #888 which fixes the naming inconsistencies.

**Comment by abalbanyan** [comment]
Any news on this? Would love to see this get merged.

**Comment by Stillness-2** [comment]
closed because #888

**Pull request closed without merging**